deploy.sh

```diff
@@ -12,29 +14,26 @@
 
 # check if resource group exist, if not proceed with creating a resource group for this deployment
 
+readonly resource_group="${1}-$(date +%s)"
+readonly location="${2:-$(az configure --list-defaults --output tsv --query "[?starts_with(name, 'location')].value")}"
+# readonly influxdb_username="${4:-admin}"
+# readonly influxdb_password="${5:-admin}"
+# readonly license_key="${LICENSE_KEY}"
 
-echo "Enter the Resource Group name:" &&
-read resourceGroupName &&
-echo "Enter the location [ie centralus, westus]:" &&
-read location &&
+if ! (az group show --name "${resource_group}"); then
+      echo -e "Creating new resource group: \"${resource_group}\"\n"
+      az group create --name "${resource_group}" --location "${location}" --output table
+fi
 
-rgvar="$((az group show --name $resourceGroupName 2>&1) | grep 'could not be found')"
-if [ -n "$rgvar" ]
-then
-      echo -e "\nThe resourceGroup $resourceGroupName does not exist and will be created\n"
-      az group create --name "${resourceGroupName}" --location "${location}" --output table
-else
-      echo -e "Confirmed resourceGroup $resourceGroupName exist and will be used\n"
+# Accept the legal terms for the influxdata offer skus (only needs ti be run once for your sunscritopn ID)
+if ! (az vm image terms show --urn influxdata:influxdb-enterprise-vm:data:1.7.10 --query "accepted"); then
+      az vm image terms accept --urn influxdata:influxdb-enterprise-vm:data:1.7.10
+      az vm image terms accept --urn influxdata:influxdb-enterprise-vm:meta:1.7.10
 fi
-# accept the legal terms for the influxdata offer skus (only needs ti be run once for your sunscritopn ID)
-echo -e "\nRuning Azure commands to accepting 'Legal Terms' for InfluxData offer datanode sku\n" 
-az vm image terms accept --urn influxdata:influxdb-enterprise-vm:data:1.7.10
-echo -e "\nRuning Azure commands to accepting 'Legal Terms' for InfluxData offer metanode sku\n" 
-az vm image terms accept --urn influxdata:influxdb-enterprise-vm:meta:1.7.10
 
 az group deployment create \
---template-file src/mainTemplate.json \
---verbose \
---resource-group "${resourceGroupName}" \
---mode Incremental \
---parameters parameters/password.parameters.json
\ No newline at end of file
+      --template-file src/mainTemplate.json \
+      --verbose \
+      --resource-group "${resource_group}" \
+      --mode Complete \
+      --parameters parameters/parameters.json
```

Comment:
might want o add the --rollback-on-error option to our default script

Reply:
this is not an urgent change, bit thought it would be worth adding since is minor and aligns with our other scripts

Reply:
I did try to add `--rollback-on-error` but it only works if there has been a successful deployment to roll back to, which doesn't exist for new deployments. 😞